planner/types.go

```diff
@@ -27,74 +27,48 @@ type PlanNode interface {
 	// Type of procedure represented by this node
 	Kind() ProcedureKind
 
-	// The types of the tables produced by this node
-	// Is it possible to know this at plan time?
-	// Type() []semantic.Type
-
 	// Helper methods for manipulating a plan
 	// These methods are used during planning
 	AddSuccessors(...PlanNode)
 	AddPredecessors(...PlanNode)
 	RemovePredecessor(PlanNode)
+	RemoveSuccessor(PlanNode)
 	ClearSuccessors()
 }
 
 type NodeID string
 
 // PlanSpec holds the result nodes of a query plan with associated metadata
 type PlanSpec struct {
-	results   map[PlanNode]struct{}
-	resources flux.ResourceManagement
-	now       time.Time
-}
-
-// CreatePlanSpec instantiates a new query plan with result nodes and metadata
-func CreatePlanSpec(results []PlanNode, resources flux.ResourceManagement, now time.Time) *PlanSpec {
-	plan := newPlanSpec(results)
-	plan.resources = resources
-	plan.now = now
-	return plan
-}
-
-func newPlanSpec(results []PlanNode) *PlanSpec {
-	r := make(map[PlanNode]struct{}, len(results))
-	for _, root := range results {
-		r[root] = struct{}{}
-	}
-	return &PlanSpec{results: r}
+	Roots     map[PlanNode]struct{}
```

Comment:
So `Roots` and `Results` are very similar.  They are likely to be confused as this is now.  Are you thinking that once the physical plan and logical plan are represented via different structures, `Roots` goes in the logical and `Results` in the physical?

Reply:
Yes exactly. In fact, I was thinking about just doing that with this PR but I didn't want to complicate things. But I can do that now if you think it's best.